lib/options.go

```diff
@@ -44,6 +45,79 @@ const DefaultScenarioName = "default"
 // nolint: gochecknoglobals
 var DefaultSummaryTrendStats = []string{"avg", "min", "med", "max", "p(90)", "p(95)"}
 
+type DNSConfig struct {
+	TTL      null.String     `json:"ttl"`
+	Strategy NullDNSStrategy `json:"strategy"`
+}
+
+//go:generate enumer -type=DNSStrategy -transform=kebab -trimprefix DNS -output dns_strategy_gen.go
+type DNSStrategy uint8
+
+type NullDNSStrategy struct {
+	DNSStrategy
+	Valid bool
+}
+
+const (
+	DNSFirst DNSStrategy = iota + 1
+	DNSRoundRobin
+	DNSRandom
+)
+
+// nolint: gochecknoglobals
+var DefaultDNSConfig = DNSConfig{
+	TTL:      null.StringFrom("inf"),
+	Strategy: NullDNSStrategy{DNSFirst, true},
+}
+
+func (c *DNSConfig) UnmarshalJSON(data []byte) error {
+	params := make(map[string]interface{})
+	if err := json.Unmarshal(data, &params); err != nil {
+		return err
+	}
+	return c.unmarshalDNSConfig(params)
+}
+
+func (c *DNSConfig) UnmarshalText(text []byte) error {
+	params, err := strvals.Parse(string(text))
+	if err != nil {
+		return err
+	}
+	return c.unmarshalDNSConfig(params)
+}
+
+func (c *DNSConfig) unmarshalDNSConfig(params map[string]interface{}) error {
+	if ttl, ok := params["ttl"]; ok {
+		c.TTL = null.StringFrom(fmt.Sprintf("%v", ttl))
+	}
+	c.Strategy = DefaultDNSConfig.Strategy
+	if strat, ok := params["strategy"]; ok {
+		if s, err := DNSStrategyString(strat.(string)); err != nil {
+			return err
+		} else {
+			c.Strategy.DNSStrategy = s
+			c.Strategy.Valid = true
+		}
+	}
```

Comment:
can we error out on unknown fields as well?

Reply:
I was refactoring `unmarshal()` anyway to see if I can get rid of the `*c = DefaultDNSConfig()` thing, so this was trivial to add. See acdf402b.